feat(navigation): add shopify driver (#4030)
graycoreio/daffodil: libs/navigation/guides/driver/shopify.md

# Shopify

> The `@daffodil/navigation` Shopify driver provides the connections between your storefront's navigation menus and your underlying Shopify store, using the Shopify Storefront API to retrieve collection-based navigation structures.

Reviewer: ```suggestion
The `@daffodil/navigation` Shopify driver connects your storefront's navigation menus to your Shopify store using the Shopify Storefront API to retrieve collection-based navigation structures.
```

Author: 9465acba0

## Features

> - **Collection-based Navigation**: Builds navigation trees from Shopify collections

Reviewer: ```suggestion
- **Collection-based navigation**: Builds navigation trees from Shopify collections
```

Author: 9465acba0

> - **GraphQL Integration**: Uses Shopify's Storefront API for efficient data fetching

Reviewer: ```suggestion
- **GraphQL integration**: Uses Shopify Storefront API for efficient data fetching
```

Author: 9465acba0

## Prerequisites

Before using the Shopify navigation driver, ensure you have:

> 1. A Shopify store with the Storefront API enabled

Reviewer: ```
- A Shopify store with Storefront API enabled
- A valid Storefront API access token
- Collections configured in your Shopify admin
```

Author: 9465acba0
